### DISCUSSION TITLE
Resolver seems to mutate payload ?

### DISCUSSION THREAD

**Discussion by mdartic** [feathersjs/feathers#3142] [Q&A]
I'm writing some tests on a service, that have resolvers.

```ts
// myservice.resolver.ts
export const myServiceCreateResolver = resolve<MyServiceResult, HookContext>({
  myProp3: async (value) => {
    return "myValue3"
  }
})

// myservice.test.ts
// inside a test :
const payload = {
  myProp1: "myValue1",
  myProp2: "myValue2"
}

const result = await app.service('myservice').create(payload)

console.log(payload) // { myProp1: "myValue1", myProp2: "myValue2", myProp3: "myValue3" }

```

This seems a bit strange for me, I would have thought `payload` was immutable.

I encounter some issues when giving the same payload through several calls to the service, as my validator breaks if `myProp3` is unwanted.

I fix it by destructuring the payload with `.create({...payload})`, but I don't have a nested payload so it's simple.

Do you think this is ok to have a mutable payload ? Or wouldn't it be better to have an immutable payload ?

**Comment by daffl** [comment]
I am not able to reproduce this. Doing something like this:

```ts
  const u: UserData = {
    email: 'test@thing.com',
    password: 'supersecret'
  }

  console.log(await app.service('users').create(u))
  console.log(u)
```

In the [feathers-chat](https://github.com/feathersjs/feathers-chat) leaves the original object unmodified.
